Taking into account the `@Priority` annotation when registering context providers

Sgitario opened the pull request:
At the moment, users can register multiple context providers for handing the same class by using: 

```java
Client client = QuarkusRestClientBuilder.newBuilder()
                    .baseUri(URI.create(uri))
                    .register(MyLowPriorityContextProvider.class)
                    .register(MyHighPriorityContextProvider.class)
                    .build(Client.class);
```

When doing this, REST Client Reactive will always return the first registered context provider for the matching type:

```java
@Priority(2)
    public static class MyLowPriorityContextProvider implements ContextResolver<Person> {

        @Override
        public Person getContext(Class<?> aClass) {
// ...
        }
    }

    @Priority(1)
    public static class MyHighPriorityContextProvider implements ContextResolver<Person> {

        @Override
        public Person getContext(Class<?> aClass) {
// ...
        }
    }
```

With these changes, the `@Priority` annotation is taking into account to return the correct context provider regardless to when it has been registered.

geoand commented:
Are you sure that the a larger number means higher priority? In JAX-RS, it's usually the other way around (unless we are talking about @Provider classes on the write path).

Perhaps we should make a check with `quarkus-rest-client` and see how it behaves?

Sgitario commented:
> Are you sure that the a larger number means higher priority? In JAX-RS, it's usually the other way around (unless we are talking about @Provider classes on the write path).
> 
> Perhaps we should make a check with `quarkus-rest-client` and see how it behaves?

You're right! Thanks for spotting.

geoand commented:
🙏🏼

Sgitario committed:
Taking into account the `@Priority` annotation in context providers

At the moment, users can register multiple context providers for handing the same class by using: 

```java
Client client = QuarkusRestClientBuilder.newBuilder()
                    .baseUri(URI.create(uri))
                    .register(MyLowPriorityContextProvider.class)
                    .register(MyHighPriorityContextProvider.class)
                    .build(Client.class);
```

When doing this, REST Client Reactive will always return the first registered context provider for the matching type:

```java
@Priority(2)
    public static class MyLowPriorityContextProvider implements ContextResolver<Person> {

        @Override
        public Person getContext(Class<?> aClass) {
// ...
        }
    }

    @Priority(1)
    public static class MyHighPriorityContextProvider implements ContextResolver<Person> {

        @Override
        public Person getContext(Class<?> aClass) {
// ...
        }
    }
```

With these changes, the `@Priority` annotation is taking into account to return the correct context provider regardless to when it has been registered.

Sgitario commented:
PR updated

Sgitario commented:
The failures are unrelated to these changes.
